Modelica/Electrical/Analog/Ideal/ImprovedOpAmpLimited.mo

```diff
@@ -0,0 +1,110 @@
+within Modelica.Electrical.Analog.Ideal;
+model ImprovedOpAmpLimited "Improved operational amplifier with limitation"
+  parameter Real V0=15000.0 "No-load amplification";
+  parameter SI.Time Tr=1e-5 "Rise time 10%-90% of output voltage";
+  parameter Boolean useSupply=false
+    "Use supply pins (otherwise constant supply)" annotation (Evaluate=true);
+  parameter SI.Voltage Vps=+15 "Positive supply voltage"
+    annotation (Dialog(enable=not useSupply));
+  parameter SI.Voltage Vns=-15 "Negative supply voltage"
+    annotation (Dialog(enable=not useSupply));
+  parameter Modelica.Electrical.Analog.Types.InitOpAmp initOpAmp=Modelica.Electrical.Analog.Types.InitOpAmp.Linear
+    "Initialization of rise time fristOrder"
+    annotation (Evaluate=true, Dialog(group="Initialization"));
+  Boolean satPos=v_int>=vps "Indicates positive Saturation";
+  Boolean satNeg=v_int<=vns "Indicates negative Saturation";
+  SI.Voltage vps "Positive supply voltage";
+  SI.Voltage vns "Negative supply voltage";
+  SI.Voltage v_in(start=0)=in_p.v - in_n.v "Input voltage difference";
+  SI.Voltage v_int "Intermediate voltage";
+  SI.Voltage v_out=out.v "Output voltage to ground";
+  SI.Current i_out(start=0)=out.i "Output current into OpAmp";
+  SI.Power p_in=in_p.v*in_p.i + in_n.v*in_n.i "Input power";
+  SI.Power p_out=out.v*out.i "Output power";
+  SI.Power p_s=-(p_in + p_out) "Supply power";
+  SI.Current i_s=p_s/(vps - vns) "Supply current";
+  Modelica.Electrical.Analog.Interfaces.PositivePin in_p
+    "Positive pin of the input port" annotation (Placement(transformation(
+          extent={{-90,-70},{-110,-50}})));
+  Modelica.Electrical.Analog.Interfaces.NegativePin in_n
+    "Negative pin of the input port" annotation (Placement(transformation(
+          extent={{-110,50},{-90,70}})));
+  Modelica.Electrical.Analog.Interfaces.PositivePin out
+    "Pin of the output port" annotation (Placement(transformation(extent={{
+            110,-10},{90,10}}), iconTransformation(extent={{110,-10},
+            {90,10}})));
+  //optional supply pins
+  Modelica.Electrical.Analog.Interfaces.PositivePin s_p(final i=+i_s, final v=vps)
+    if useSupply "Optional positive supply pin" annotation (Placement(
+        transformation(extent={{10,90},{-10,110}})));
+  Modelica.Electrical.Analog.Interfaces.NegativePin s_n(final i=-i_s, final v=vns)
+    if useSupply "Optional negative supply pin" annotation (Placement(
+        transformation(extent={{-10,-110},{10,-90}})));
+initial equation
+  if initOpAmp == Modelica.Electrical.Analog.Types.InitOpAmp.UpperLimit then
+    v_int=vps;
+  elseif initOpAmp == Modelica.Electrical.Analog.Types.InitOpAmp.LowerLimit
+       then
+    v_int=vns;
+  else
+    v_int=V0*v_in;
+  end if;
+equation
+  if not useSupply then
+    vps = Vps;
+    vns = Vns;
+  end if;
+  assert(vps > vns, "Supply voltages are not consistent: vps <= vns");
```

Comment:
There's no longer any need to repeat the condition; see https://github.com/modelica/ModelicaSpecification/pull/3641

Reply:
Sorry I missed that discussion. Accepted.